src/serac/physics/tests/serac_thermal_mechanics_functional.cpp

```diff
@@ -0,0 +1,397 @@
+
+  // Define a boundary attribute set
+  std::set<int> constraint_bdr = {1, 2, 3, 4, 5, 6};
+  std::set<int> temp_bdr       = {1, 2, 3, 4, 5, 6};
+
+  // Set a constant initial temperature, at the thermal expansion reference temperature value.
+  auto initial_temperature_field = [theta_ref](const mfem::Vector&, double) -> double { return theta_ref; };
+  thermal_solid_solver.setTemperature(initial_temperature_field);
+
+  // set a constant temperature boundary condition
+  // set it below the reference temperature
+  auto one = [theta_ref](const mfem::Vector&, double) -> double { return theta_ref - 0.0; };
+  thermal_solid_solver.setTemperatureBCs(temp_bdr, one);
+
+  // displacement boundary condition
+  thermal_solid_solver.setDisplacementBCs(constraint_bdr, exact_displacement_function);
+
+  // Cheating initial guess
+  auto near_exact_displacement_function = [&A, &b](const mfem::Vector& X, mfem::Vector& u) {
+    A.Mult(X, u);
+    u += b;
+    u *= 0.99;
+  };
+  thermal_solid_solver.setDisplacement(near_exact_displacement_function);
+
+  // double G = 0.5*E/(1.0 + nu);
+  // double K = E/3.0/(1.0 - 2.0*nu);
+  // thermal_solid_solver.addBodyForce(
+  //     [K, G, alpha0](const tensor<double, 3>& /*x*/, const double /*t*/, const auto& /*u*/, const auto& /*dudx*/) {
+  //       double dummy = K*G*alpha0;
+  //       return tensor<double, 3>{{0.0, 0.0, 0.0}};
+  //     });
+  //{{6.0*K*((1.0 - x[0]*0.5)*alpha0 - 0.5*x[0]) - 8.0/3.0*G - 2.0*K, 0.0, 0.0}};
+
+  thermal_solid_solver.completeSetup();
+
+  // thermal_solid_solver.initializeOutput(serac::OutputType::VisIt, "thermal_mechanics_without_input_file_output");
+  // dump initial state to output
+  thermal_solid_solver.outputState("pv_output");
+
+  for (int i = 0; i < 4; i++) {
+    // Perform the quasi-static solve
+    double dt = 1.0;
+    thermal_solid_solver.advanceTimestep(dt);
+
+    thermal_solid_solver.outputState();
+
+    // Compute norm of error in numerical solution
+    // auto exact_solution_coef = std::make_shared<mfem::VectorFunctionCoefficient>(
+    //     dim, exact_displacement_function);
+    mfem::VectorFunctionCoefficient exact_solution_coef(dim, exact_displacement_function);
+    double error_norm = thermal_solid_solver.displacement().gridFunction().ComputeL2Error(exact_solution_coef);
+
+    std::cout << error_norm << std::endl;
+  }
+
+  // EXPECT_LT(error_norm, 1e-10);
+}
+
+}  // namespace serac
+
+TEST(Thermomechanics, staticTest)
+{
+  constexpr int p = 2;
+  serac::functional_test_static_3D<p>(0.0);
+}
+
+TEST(Thermomechanics, thermalContraction)
+{
+  constexpr int p = 2;
+  // this is the small strain solution, which works with a loose enought tolerance
+  // TODO work out the finite deformation solution
+  double alpha       = 1e-3;
+  double L           = 8;
+  double delta_theta = 1.0;
+  serac::functional_test_shrinking_3D<p>(std::sqrt(L * L * L / 3.0) * alpha * delta_theta);
+}
+
+TEST(Thermomechanics, parameterized)
```

Comment:
I'm not sure what this stub is a placeholder for.

Reply:
this is that "failing" test, which is currently commented out rather than completely deleted

Reply:
Yeah, I would completely delete it in favor of a github issue.